include/onnxruntime/core/framework/sparse_tensor.h

```diff
@@ -1,72 +1,427 @@
+    return ml_data_type_;
   }
 
-  //TensorShape& MutableShape() {
-  //  return shape_;
-  //}
+  /// <summary>
+  /// Test for string type
+  /// </summary>
+  /// <returns>true if tensor values are strings</returns>
+  bool IsDataTypeString() const {
+    return utils::IsPrimitiveDataType<std::string>(ml_data_type_);
+  }
+
+  /// <summary>
+  /// Checks if the Tensor contains data type T
+  /// </summary>
+  /// <typeparam name="T"></typeparam>
+  /// <returns>true if tensor contains data of type T</returns>
+  template <class T>
+  bool IsDataType() const {
+    return utils::IsPrimitiveDataType<T>(ml_data_type_);
+  }
+
+  const OrtMemoryInfo& Location() const noexcept { return location_; }
+
+  bool OwnsBuffer() const noexcept {
+    return allocator_ != nullptr;
+  }
+
+  /// <summary>
+  /// Read only access to Coo index
+  /// </summary>
+  class CooView {
+   public:
+    explicit CooView(const Tensor& index) noexcept
+        : index_(index) {}
+    const Tensor& Index() const noexcept { return index_; }
+
+   private:
+    std::reference_wrapper<const Tensor> index_;
+  };
+
+  /// <summary>
+  /// Returns Coo index view
+  /// </summary>
+  /// <returns></returns>
+  CooView AsCoo() const;
+
+  /// <summary>
+  /// Uses COO index contained the user allocated buffer along with the values buffer passed on
+  /// to the constructor. The buffer is used as is. The location of the index is assumed to be
+  /// the same as values.
+  ///
+  /// The index size must either exactly match the number of values in which case
+  /// index shape would be 1-D (values_count) or it must be twice the number of values
+  /// in which case its shape would be 2-D (values_count, 2)
+  /// </summary>
+  /// <param name="index">user allocated buffer span</param>
+  /// <returns>Status</returns>
+  Status UseCooIndex(gsl::span<const int64_t> index);
+
+  /// <summary>
+  /// The method allocates a single contiguous buffer and copies specified values
+  /// and index into it using supplied IDataTransfer
+  ///
+  /// The index size must either exactly match the number of values in which case
+  /// index shape would be 1-D (values_count) or it must be twice the number of values
+  /// in which case its shape would be 2-D (values_count, 2)
+  /// </summary>
+  /// <param name="values_count"></param>
+  /// <param name="values_data"></param>
+  /// <param name="index"></param>
+  /// <returns></returns>
+  Status MakeCooData(const IDataTransfer& data_transfer, const OrtMemoryInfo& data_location,
+                     size_t values_count, const void* values_data, gsl::span<const int64_t> index);
+
+  /// <summary>
+  /// Gives mutable access to Coo buffers so they can be populated
+  /// </summary>
+  class CooMutator {
+   public:
+    CooMutator(Tensor& values, Tensor& index) noexcept : values_(values), index_(index) {}
+    Tensor& Values() noexcept { return values_; }
+    Tensor& Index() noexcept { return index_; }
+
```

Comment:
nit: `Indexes` would be clearer and more consistent with `Values`

Reply:
same comment applies to the other places returning the Tensor of indexes from a method called `Index`

Reply:
Indices spelling was suggested last time.

Reply:
Indices is good too.